stubs/golang/build.gradle

```diff
@@ -0,0 +1,142 @@
+        proto {
+            // A source directory must be specified for protobuf-gradle-plugin to work, but since we have
+            // no sources in this repository we just set it to an empty directory.
+            srcDirs = [file('build/proto-src')]
+        }
+    }
+}
+
+protobuf {
+    protoc {
+        // Download from repositories
+        artifact = 'com.google.protobuf:protoc:3.6.0'
+    }
+    generateProtoTasks {
+        all().each {task ->
+            task.plugins {
+                go {
+                    option 'plugins=grpc'
+                }
+            }
+            task.builtins {
+                remove java
+            }
+        }
+    }
+    plugins {
+        go {
+            path = file('.gogradle/project_gopath/bin/protoc-gen-go')
+        }
+    }
+}
+
+gitPublish {
+    repoUri = 'git@github.com:infostellarinc/go-stellarstation.git'
+    branch = 'master'
+
+    preserve {
+        include '**'
+        exclude 'api'
+    }
+
+    // what to publish, this is a standard CopySpec
+    contents {
+        from 'api' into 'api'
+    }
+}
+
+task installProtocGoPlugin(type: com.github.blindpirate.gogradle.Go) {
+    go 'get -u github.com/golang/protobuf/protoc-gen-go'
+}
+
+task installProtoWrap(type: com.github.blindpirate.gogradle.Go) {
+    go 'get -u github.com/square/goprotowrap/cmd/protowrap'
+}
+
+afterEvaluate {
+    tasks.generateProto.dependsOn tasks.clean, installProtocGoPlugin, installProtoWrap
+    tasks.assemble.dependsOn tasks.generateProto
+    tasks.gitPublishReset.dependsOn tasks.generateProto
+
+    // Hack around golang protobuf plugin not supporting compiling multiple packages at the same
+    // time. We use the protowrap wrapper which does support it and hack it into the generateProto
+    // task.
+    def generateProto = tasks.generateProto
+    // We need to round-trip through either doFirst or doLast so Closure is converted to Action.
+    // We add it to the last and pop it off and insert where we want it.
+    generateProto.doLast {
+        def protocPath = protobuf.tools.protoc.path
+        def protowrapPath = file('.gogradle/project_gopath/bin/protowrap')
+        def runProtowrap = file("${buildDir}/scripts/run-protowrap.sh")
+        runProtowrap.parentFile.mkdirs()
+        runProtowrap.text = """
+#!/usr/bin/env sh
+
+${protowrapPath} --protoc_command=${protocPath} \$@
+"""
+        runProtowrap.executable = true
+        project.protobuf.tools.protoc.path = runProtowrap
+    }
+    def hackProtoWrap = generateProto.actions.pop()
+    // HACK(anuraaga): This is dependent on gradle-protobuf-plugin implementation details. Protoc
+    // resolution happens here so we insert our hack after it.
+    generateProto.actions.addAll(1, hackProtoWrap)
+
+    generateProto.doFirst {
+        mkdir file('build/proto-src')
+    }
+    generateProto.doLast {
+        copy {
+            from file('build/generated/source/proto/main/go/stellarstation')
+            into projectDir
```

Comment:
We can copy into a directory under `build` now since this is published, not the repo itself. Also you may just be able to generate straight into the repo, though if it adds complication of us having to manually delete `api` or something, don't worry about it.

https://github.com/ajoberstar/gradle-git-publish#avoiding-extra-copy